service/controller/clusterapi/v27/resource/secretfinalizer/delete.go

```diff
@@ -0,0 +1,81 @@
+package secretfinalizer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/finalizerskeptcontext"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v27/key"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// When the operator's resource implementations request the CR's finalizers to
+	// be kept, their deletion logic is delayed. That implies that we should not
+	// remove the secret finalizers here already, since certain resource
+	// implementations may still require secrets to be available during their own
+	// deletion logic execution in upcoming reconciliation loops.
+	if finalizerskeptcontext.IsKept(ctx) {
```

Comment:
I think this makes sense. We should actually do this in `app-operator` the same way I think. Question is only how the primitives change when we go with some reconcile result. /cc @kopiczko @tomahawk28

Reply:
That's clever. We need to make that doable in the future as well. This is useful.